src/inspector/inspector_tcp.cpp

```diff
@@ -30,25 +25,29 @@ void tcp_server::run() {
   io_service_.run();
 }
 
+void tcp_server::stop() {
+  boost::system::error_code ec;
+  acceptor_.close(ec);
+  socket_.close(ec);
+
+  io_service_.stop();
+}
+
 void tcp_server::do_accept()
 {
+  std::shared_ptr<tcp_server> self;
   acceptor_.async_accept(socket_,
-    [this](boost::system::error_code ec)
+    [this, self](boost::system::error_code ec)

```

Comment:
>[this, self [](start = 4, length = 11)

why is this lifetime needed here?

Reply:
As far as i remember .. the shared pointer capture is required to keep the object alive .. This is a model followed by boost::asio ... "this" capture was added as a shortcut to be able to access the private variables..

---
In reply to: [447997236](https://github.com/microsoft/v8-jsi/pull/17#discussion_r447997236) [](ancestors = 447997236)